### ISSUE TITLE
Bugs in 0.1.0

### ISSUE THREAD

**Issue by kephale** [scenerygraphics/SciView#76]
- [ ] Tried to run convex hull on Robert's shell mesh (in main/resources) and got:
```
java.lang.IndexOutOfBoundsException
	at java.nio.Buffer.checkIndex(Buffer.java:540)
	at java.nio.DirectFloatBufferU.get(DirectFloatBufferU.java:253)
	at graphics.scenery.HasGeometry$DefaultImpls.recalculateNormals(HasGeometry.kt:825)
	at graphics.scenery.Mesh.recalculateNormals(Mesh.kt:14)
	at sc.iview.process.MeshConverter.toScenery(MeshConverter.java:85)
	at sc.iview.SciView.addMesh(SciView.java:724)
	at sc.iview.ops.ConvexHull.run(ConvexHull.java:66)
	at org.scijava.command.CommandModule.run(CommandModule.java:199)
	at org.scijava.module.ModuleRunner.run(ModuleRunner.java:168)
	at org.scijava.module.ModuleRunner.call(ModuleRunner.java:127)
	at org.scijava.module.ModuleRunner.call(ModuleRunner.java:66)
	at org.scijava.thread.DefaultThreadService$3.call(DefaultThreadService.java:238)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:748)
```
- [x] When voxelizing a mesh, 2 images are shown instead of 1.

- [x] When loading the shell mesh, then voxelizing it, and reimporting as isosurface, I got:
```
java.lang.ClassCastException: net.imglib2.type.numeric.integer.UnsignedByteType cannot be cast to net.imglib2.type.logic.BitType
	at net.imglib2.type.logic.BitType.compareTo(BitType.java:55)
	at net.imagej.ops.threshold.apply.ApplyThresholdComparable.compute(ApplyThresholdComparable.java:56)
	at net.imagej.ops.threshold.apply.ApplyThresholdComparable.compute(ApplyThresholdComparable.java:46)
	at net.imagej.ops.special.computer.BinaryComputerOp.compute(BinaryComputerOp.java:93)
	at net.imagej.ops.map.MapIterableToIterable.compute(MapIterableToIterable.java:58)
	at net.imagej.ops.map.MapIterableToIterable.compute(MapIterableToIterable.java:47)
	at net.imagej.ops.threshold.apply.ApplyConstantThreshold.compute(ApplyConstantThreshold.java:73)
	at net.imagej.ops.threshold.apply.ApplyConstantThreshold.compute(ApplyConstantThreshold.java:50)
	at net.imagej.ops.threshold.AbstractApplyThresholdIterable.compute(AbstractApplyThresholdIterable.java:52)
	at net.imagej.ops.threshold.AbstractApplyThresholdIterable.compute(AbstractApplyThresholdIterable.java:43)
	at net.imagej.ops.special.hybrid.UnaryHybridCF.calculate(UnaryHybridCF.java:61)
	at net.imagej.ops.special.hybrid.UnaryHybridCF.run(UnaryHybridCF.java:71)
	at net.imagej.ops.special.hybrid.UnaryHybridCF.run(UnaryHybridCF.java:97)
	at org.scijava.command.CommandModule.run(CommandModule.java:199)
	at net.imagej.ops.OpEnvironment.run(OpEnvironment.java:944)
	at net.imagej.ops.OpEnvironment.run(OpEnvironment.java:156)
	at net.imagej.ops.threshold.ThresholdNamespace.apply(ThresholdNamespace.java:82)
	at sc.iview.io.ImportIsosurface.run(ImportIsosurface.java:67)
	at org.scijava.command.CommandModule.run(CommandModule.java:199)
	at org.scijava.module.ModuleRunner.run(ModuleRunner.java:168)
	at org.scijava.module.ModuleRunner.call(ModuleRunner.java:127)
	at org.scijava.module.ModuleRunner.call(ModuleRunner.java:66)
	at org.scijava.thread.DefaultThreadService$3.call(DefaultThreadService.java:238)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:748)
```

Edit @skalarproduktraum: Changed to task list

**Comment by kephale** [comment]
Screenshots are upside down. IIRC there is something about flipping the scenery buffer when it is being loaded into a JavaFX window.

**Comment by kephale** [comment]
1) The convex hull error comes from a missing normals issue. This is resolved in https://github.com/imagej/imagej-mesh/tree/recalculate-normals

**Comment by kephale** [comment]
2. Fixed in https://github.com/scenerygraphics/SciView/commit/69aed152d5ade56d5d6d4d868a78964a46f3644a

**Comment by skalarproduktraum** [comment]
The screenshot issue should be fixed with https://github.com/scenerygraphics/scenery/commit/31e7dd205c57a4865c47dcd9cc496598d4392c87 which also went into 0.4.2 ... can you confirm?

**Comment by kephale** [comment]
Yeah screenshots are fixed in the current release. Once I resolve the voxelization issue, which is simple-ish but requires some imglib2 type checking, i'll close this thread

**Comment by kephale** [comment]
3) is solved in the current version of master
1) is finished and waiting on an imagej-mesh release before https://github.com/scenerygraphics/SciView/tree/fix-convex-hull can get merged into master

jobs done
